src/Morpho.sol

```diff
@@ -419,10 +419,12 @@ contract Morpho is IMorphoStaticTyping {
 
     /// @inheritdoc IMorphoBase
     function flashLoan(address token, uint256 assets, bytes calldata data) external {
-        IERC20(token).safeTransfer(msg.sender, assets);
+        require(assets != 0, ErrorsLib.ZERO_ASSETS);
 
         emit EventsLib.FlashLoan(msg.sender, token, assets);
```

Comment:
I believe this event was purposefully emitted before the first transfer

But I'm fine with it being emitted before too

Reply:
it's easier if everywhere we emit before transfers no?

because now there is no clear rule